NavigationReactNative/src/android/src/main/java/com/navigation/reactnative/ToolbarView.java

```diff
@@ -201,20 +223,28 @@ public boolean onMenuItemActionExpand(MenuItem item) {
                 }
             }
         }
-        setMenuTintColor();
+        setMenuTintColor(testIDs);
         requestLayout();
     }
 
-    private void setMenuTintColor()  {
+    private void setMenuTintColor(HashMap<MenuItem, String> testIDs)  {
         for (int i = 0; i < getChildCount(); i++) {
             if (getChildAt(i) instanceof ActionMenuView) {
                 ActionMenuView menu = (ActionMenuView) getChildAt(i);
-                for(int j = 0; j < menu.getChildCount(); j++) {
+                for (int j = 0; j < menu.getChildCount(); j++) {
                     if (menu.getChildAt(j) instanceof ActionMenuItemView) {
-                        ActionMenuItemView menuItem = (ActionMenuItemView) menu.getChildAt(j);
+                        ActionMenuItemView menuItemView = (ActionMenuItemView) menu.getChildAt(j);
                         if (defaultMenuTintColor == null)
-                            defaultMenuTintColor = menuItem.getCurrentTextColor();
-                        menuItem.setTextColor(tintColor != null ? tintColor : defaultMenuTintColor);
+                            defaultMenuTintColor = menuItemView.getCurrentTextColor();
+                        menuItemView.setTextColor(tintColor != null ? tintColor : defaultMenuTintColor);
+                        if (testIDs != null) {
+                            MenuItem menuItem = ((MenuView.ItemView) menuItemView).getItemData();
+                            menuItemView.setTag(testIDs.get(menuItem));
+                        }
```

Comment:
This is another place where we need to update the button testIDs if someone changes them after the first render (see comment above about navigation and overflow testID). Let's take a shortcut here and fix this in the `BarButton` React component instead. If we set a key that's based on the testID then when the testID changes the `BarButton` will be removed and re-added so it will go into `setMenuItems` again. For safety, let's set the key based on the testID, show and title - so that if any of them change then the setMenuItems is called. Notice below that I only set the key on android if the testID is populated 
```jsx
<NVBarButton
    key={(Platform.OS === 'ios' ? systemItem : (testID ? testID + show + (title || '') : null))}
```